src/pages/docs/[[...slug]]/index.tsx

```diff
@@ -50,12 +57,33 @@ export async function getStaticPaths() {
 export async function getStaticProps({ params }) {
   const { slug } = params
 
-  const doc = allDocs.find(doc => doc.slugAsParams === slug.join('/'))
+  const index = allDocs.findIndex((doc, index) => doc.slugAsParams === slug.join('/'))
 
-  if (!doc) {
+  if (index === undefined) {
     return { notFound: true }
   }
 
+  const isComponentDoc = doc => doc.slugAsParams.split('/').length === 1
+  const isCurrentDoc = currentDoc => doc =>
+    doc.slugAsParams.split('/').slice(1) === currentDoc.slugAsParams.split('/').slice(1)
+
+  const currentDoc = allDocs[index]
+  const isCurrent = isCurrentDoc(currentDoc)
+
+  const doc = Object.assign(currentDoc, {
+    prev:
+      allDocs
+        .slice(0, index)
+        .filter(doc => !isCurrent(doc))
+        .reverse()
+        .find(isComponentDoc)?.slugAsParams || null,
+    next:
+      allDocs
+        .slice(index + 1)
+        .filter(doc => !isCurrent(doc))
+        .find(isComponentDoc)?.slugAsParams || null,
```

Comment:
There is not need of doing this IMO. You can just pass `prev` and `next` as props to the page. This way you don't change the type of the doc component

Reply:
It fails the typechecking because the Doc object does not include it

Reply:
understood

Reply:
gonna change it